saleor/graphql/notifications/tests/test_external_notification_query.py

```diff
@@ -0,0 +1,156 @@
+import json
+from unittest.mock import patch
+
+import pytest
+from graphql_relay.node.node import to_global_id
+
+from ....account.models import User
+from ....core.notify_events import UserNotifyEvent
+from ....graphql.tests.utils import assert_no_permission
+from ....plugins.tests.sample_plugins import PluginSample
+
+query_test_data = [
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            },
+            "pluginId": "",
+        },
+        200,
+    ),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            },
+            "pluginId": "WRONG-TEST-PLUGIN",
+        },
+        200,
+    ),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            }
+        },
+        200,
+    ),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+            }
+        },
+        400,
+    ),
+    (
+        {
+            "input": {
+                "extraPayload": json.dumps("{}"),
+                "externalEventType": {},
+                "channel": "c-pln",
+            }
+        },
+        400,
+    ),
+    ({"input": {"ids": [], "externalEventType": {}, "channel": "c-pln"}}, 200),
+    (
+        {
+            "input": {
+                "ids": [],
+                "extraPayload": json.dumps("{}"),
+                "channel": "c-pln",
+            }
+        },
+        400,
+    ),
+]
+
+
+@pytest.mark.parametrize("variables, status_code", query_test_data)
+def test_query(
```

Comment:
Wouldn't be good to have `test_external_notification_trigger_query` ?

Reply:
Also, I am not convinced about this test.  It tests response codes that with graphql are not really important (you can receive 200 with error from API). 
Instead of this, you should do something like this:
1st test for incorrect arguments and checking that API returned errors (something similar to this):
```
    content = get_graphql_content(response)
    data = content["data"]["appInstall"]

    errors = data["errors"]
    assert not data["appInstallation"]
    assert len(errors) == 1
    error = errors[0]
    assert error["field"] == "permissions"
    assert error["code"] == AppErrorCode.OUT_OF_SCOPE_PERMISSION.name
    assert error["permissions"] == [PermissionEnum.MANAGE_ORDERS.name]
```

2nd test - correct input - it is already covered by below tests